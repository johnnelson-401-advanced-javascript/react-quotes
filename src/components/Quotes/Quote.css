.Quote {
  font-size: 2em;
}