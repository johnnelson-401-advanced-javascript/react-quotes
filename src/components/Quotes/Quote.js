import React from 'react';
import PropTypes from 'prop-types';
import styles from './Quote.css';

const Quote = ({ image, character, quote }) => (
  <figure className={styles.Quote}>
    <img src={image} alt={character} />
    <figcaption>{character}</figcaption>
    <figcaption>{quote}</figcaption>
  </figure>
);

Quote.propTypes = {
  image: PropTypes.string.isRequired,
  character: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired,
};

export default Quote;
