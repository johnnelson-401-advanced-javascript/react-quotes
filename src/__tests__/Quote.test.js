import React from 'react';
import { shallow } from 'enzyme';
import Quote from '../components/Quotes/Quote';

describe('Renders a Quote Component', () => {
  it('Tests the Quote component', () => {
    const quote = { image: 'string.com', character: 'Name', quote: 'Test' };
    const handleClick = jest.fn();
    const wrapper = shallow(<Quote quote={quote} handleClick={handleClick} />);
    expect(wrapper).toMatchSnapshot();


  });  
});
