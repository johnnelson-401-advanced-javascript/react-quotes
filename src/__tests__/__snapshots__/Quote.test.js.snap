// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a Quote Component Tests the Quote component 1`] = `
<figure
  className="Quote"
>
  <img
    alt="Name"
    src="string.com"
  />
  <figcaption>
    Name
  </figcaption>
  <figcaption>
    <Component />
  </figcaption>
  <button
    onClick={[MockFunction]}
  >
    Get New Quote
  </button>
</figure>
`;
